File: flutter_module/lib/structs/changenotifierprovider_struct.dart
// FFI struct for ChangeNotifierProvider widget
// ChangeNotifierProvider manages a ChangeNotifier and notifies consumers
// when it changes. The ChangeNotifier is managed on C# side.

import 'dart:ffi';
import 'package:ffi/ffi.dart';
import '../flutter_sharp_structs.dart';

/// FFI struct representation of ChangeNotifierProvider widget.
final class ChangeNotifierProviderStruct extends Struct {
  // FlutterObject Struct fields
  external Pointer handle;
  external Pointer managedHandle;
  external Pointer<Utf8> widgetType;

  // WidgetStruct field
  external Pointer<Utf8> id;

  /// Pointer to the child widget tree.
  external Pointer<WidgetStruct> child;
}

File: src/Flutter/Structs/ChangeNotifierProviderStruct.cs
using System;
using System.Runtime.InteropServices;

namespace Flutter.Structs
{
    /// <summary>
    /// FFI struct for ChangeNotifierProvider widget.
    /// </summary>
    /// <remarks>
    /// ChangeNotifierProvider manages a ChangeNotifier and notifies
    /// consumers when it changes. The ChangeNotifier itself is managed
    /// on the C# side - this struct only contains the child widget pointer.
    /// </remarks>
    [StructLayout(LayoutKind.Sequential)]
    internal class ChangeNotifierProviderStruct : WidgetStruct
    {
        /// <summary>
        /// Pointer to the child widget tree.
        /// </summary>
        public IntPtr child;
    }
}
